### PR TITLE
Add delay seconds to publish.

### PR DESCRIPTION
**Pull request by kkaehler**
@wenbinf

### CODE REVIEW

File: kale/task.py [hunk 1]
```diff
@@ -88,15 +88,24 @@ def _get_task_id(cls, *args, **kwargs):
         return '%s_uuid_%s' % (cls.__name__, uuid.uuid1())
 
     @classmethod
-    def publish(cls, app_data, *args, **kwargs):
+    def publish(cls, *args, **kwargs):
         """Class method to publish a task given instance specific arguments."""
+        return cls.publish_with_delay(None, *args, **kwargs)
```
Comment (mrterry) [inline kale/task.py]: `None` or `0`?

File: kale/task.py [hunk 2]
```diff
@@ -88,15 +88,24 @@ def _get_task_id(cls, *args, **kwargs):
...
+
+    @classmethod
+    def publish_with_delay(cls, delay_sec, app_data, *args, **kwargs):
+        """Class method to publish a task with a delay.
+
+        Args:
+            delay_sec: Number of seconds to delay the task.
```
Comment (mrterry) [inline kale/task.py]: If delay_sec is `None` what happens?

Comment (kkaehler) [inline kale/task.py]: we use the default which is 0

File: kale/task.py (hunk 1 above)
Comment (kkaehler) [inline kale/task.py]: I prefer none because then we fall back on the default (which is also 0)

File: kale/task.py (hunk 2 above)
Comment (mrterry) [inline kale/task.py]: This is a strange beast: A required arg that happens to have (unspecified) default behavior if you pass in None.  Can you at least update the docs to reflect this?

    delay_sec: Number of seconds to delay the task.  If None, publish with the default delay (probably 0).

**Comment by wenbinf** [comment]
This change looks trivial and simple, but it turns out to be very tricky. It's function signature change. It needs thorough testing. 

As discussed offline with kip, we decided not to do this for now.

**Pull request closed without merging**